src/benchmark_ocl.cpp

```diff
@@ -144,125 +119,106 @@ int main(void)
+	noma::ocl::helper ocl_helper(ocl_config);
 	// output the used device
-	cl_device_id dev_id;
-	err = clGetCommandQueueInfo(CLU_DEFAULT_Q, CL_QUEUE_DEVICE, sizeof(cl_device_id), &dev_id, nullptr);
-	ocl_error_handler(err, "clGetCommandQueueInfo()");
-	clu_device_info dev_info = cluGetDeviceInfo(dev_id, &err);
-	ocl_error_handler(err, "cluGetDeviceInfo()");
-	std::cerr << "OpenCL device name:    " << dev_info.device_name << std::endl;
-	std::cerr << "OpenCL device vendor:  " << dev_info.device_vendor << std::endl;
-	std::cerr << "OpenCL device profile: " << dev_info.device_profile << std::endl;
-	std::cerr << "OpenCL device version: " << dev_info.device_version << std::endl;
-	std::cerr << "OpenCL driver version: " << dev_info.driver_version << std::endl;
-
-	// allocate OpenCL device memory
-	cl_mem hamiltonian_ocl = clCreateBuffer(CLU_CONTEXT, CL_MEM_READ_ONLY, size_hamiltonian_byte, 0, &err);
-	ocl_error_handler(err, "clCreateBuffer(hamiltonian_ocl)");
-	cl_mem sigma_in_ocl = clCreateBuffer(CLU_CONTEXT, CL_MEM_READ_WRITE, size_sigma_byte, 0, &err);
-	ocl_error_handler(err, "clCreateBuffer(sigma_in_ocl)");
-	cl_mem sigma_out_ocl = clCreateBuffer(CLU_CONTEXT, CL_MEM_READ_WRITE, size_sigma_byte, 0, &err);
-	ocl_error_handler(err, "clCreateBuffer(sigma_out_ocl)");
+	ocl_helper.write_device_info(std::cerr);
+
+	// allocate OpenCL device memory // DONE: replace with create_buffer() from noma::ocl, use C++ interface
+	noma::ocl::buffer hamiltonian_ocl = ocl_helper.create_buffer(CL_MEM_READ_ONLY,  size_hamiltonian_byte);
+	noma::ocl::buffer sigma_in_ocl    = ocl_helper.create_buffer(CL_MEM_READ_WRITE, size_sigma_byte      );
+	noma::ocl::buffer sigma_out_ocl   = ocl_helper.create_buffer(CL_MEM_READ_WRITE, size_sigma_byte      );
 
 	// function to build and set-up a kernel
 	auto prepare_kernel = [&](const std::string& file_name, const std::string& kernel_name, const std::string& compile_options)
 	{
 		// build kernel
 		time::rep build_time = 0.0;
-		cl_program prog;
+		cl::Program prog;
 		{
 			time::timer t;
-			prog = cluBuildSourceFromFile(file_name.c_str(), compile_options.c_str(), &err);
+			prog = ocl_helper.create_program_from_file(file_name, "", compile_options);
 			build_time = t.elapsed();
 		}
 		
 		std::stringstream time_ss;
 		time_ss << std::scientific << build_time;
 		std::cout << "build_time\t" << kernel_name << "\t" << time_ss.str() << std::endl;
 		
-		if (ocl_error_handler(err, "cluBuildSourceFromFile()", false))
-		{
-			std::cerr << "OpenCL build log for: " << file_name << std::endl
-			          << cluGetBuildErrors(prog) << std::endl
-			          << "--- end of build log ---" << std::endl;
-			exit(-1);
-		}
-		cl_kernel kernel = clCreateKernel(prog, kernel_name.c_str(), &err);
-		ocl_error_handler(err, "clCreateKernel()");
+		// get kernel from programm using C++ OCL API
+		cl::Kernel kernel(prog, kernel_name.c_str(), &err);
+		noma::ocl::error_handler(err, "Error creating kernel.");
```

Comment:
Let's add the kernel name for easier debugging:
`noma::ocl::error_handler(err, "Error creating kernel: '" + kernel_name + "'.");`

Reply:
Done with a758cba0fa201d64c477ff298ffa7ea784117a09.